test/unit/modules/accounts.js

```diff
@@ -147,6 +150,64 @@ describe('accounts', () => {
 		});
 	});
 
+	describe('setAccountAndGet', () => {
+		it('should fail if address and publicKey is missing', done => {
+			const account = accountFixtures.Account();
+
+			delete account.address;
+			delete account.publicKey;
+
+			accounts.setAccountAndGet(account, (error, data) => {
+				expect(error).to.be.eql('Invalid public key');
+				expect(data).to.be.undefined;
+				done();
+			});
+		});
+
+		it('should be ok with address but no publicKey', done => {
+			const account = accountFixtures.Account();
+
+			delete account.publicKey;
+
+			accounts.setAccountAndGet(account, (error, data) => {
+				expect(error).to.be.null;
+				expect(data.address).to.be.eql(account.address);
+				expect(data.publicKey).to.not.exist;
+				done();
+			});
+		});
+
+		it('should set address if it is not provided', done => {
+			const account = accountFixtures.Account();
+
+			delete account.address;
+
+			accounts.setAccountAndGet(account, (error, data) => {
+				expect(error).to.be.null;
+				expect(data.publicKey).to.be.eql(account.publicKey);
+				expect(data.address).to.exist;
+				done();
+			});
+		});
+
+		it('should be ok with pre existing database transaction', done => {
```

Comment:
Please be more specific in tests expectations: "should be ok" => "should return no error and address"